src/pages/workspaces/workspace/Dashboard/Dashboard.ts

```diff
@@ -104,261 +66,42 @@ interface WorkspaceDashboardProps {
   workspace: Workspace;
 }
 
-const WorkspaceDashboard = forwardRef((props: WorkspaceDashboardProps, ref: ForwardedRef<unknown>): ReactNode => {
+export const WorkspaceDashboard = forwardRef((props: WorkspaceDashboardProps): ReactNode => {
   const {
     namespace,
     name,
     refreshWorkspace,
     storageDetails,
     workspace,
     workspace: {
-      accessLevel,
       owners = [],
       workspace: { authorizationDomain, attributes = { description: '' } },
     },
   } = props;
 
-  const description = attributes.description;
-
-  // State
-  const [storageCost, setStorageCost] = useState<{ isSuccess: boolean; estimate: string; lastUpdated?: string }>();
-  const [bucketSize, setBucketSize] = useState<{ isSuccess: boolean; usage: string; lastUpdated?: string }>();
-  const [editDescription, setEditDescription] = useState<string>();
-  const [saving, setSaving] = useState<boolean>(false);
-  const [busy, setBusy] = useState<boolean>(false);
-  const [tagsList, setTagsList] = useState<string[]>();
-  const [acl, setAcl] = useState<WorkspaceAcl>();
-
   const persistenceId = `workspaces/${namespace}/${name}/dashboard`;
 
-  const signal = useCancellation();
-
-  const refresh = () => {
```

Comment:
This is no longer necessary, because `refresh` is only called after the `WorkspaceContainer` calls `refreshWorkspace` from `useWorkspace`, and `refreshWorkspace` will update the workspace, triggering all the hooks that effects that depend on it.
The components that were broken out that now contain the state that was fetched here all have their `useEffect` hooks properly set up.